File: src/config/env.rs
use std::env;

pub struct Config {
    pub port: u16,
    pub api_key: String,
    pub mongodb_uri: String,
    pub mongodb_db_name: String,
}

impl Config {
    pub fn load() -> Self {
        dotenvy::dotenv().ok();

        let port = env::var("PORT")
            .ok()
            .and_then(|s| s.parse::<u16>().ok())
            .unwrap_or(8080);

        let api_key = env::var("API_KEY").unwrap_or_else(|_| "your_api_key_here".to_string());

        let mongodb_uri =
            env::var("MONGODB_URI").unwrap_or_else(|_| "mongodb://localhost:27017".to_string());

        let mongodb_db_name =
            env::var("MONGODB_DB_NAME").unwrap_or_else(|_| "mydatabase".to_string());

        Self {
            port,
            api_key,
            mongodb_uri,
            mongodb_db_name,
        }
    }
}

File: src/config/env.ts
export const ENV = {
  IS_DEV: Bun.env.NODE_ENV != "production",
  PORT: Bun.env.PORT || "3100",
  API_KEY: Bun.env.API_KEY || "secret-api-key",
  MONGODB_URI: Bun.env.MONGODB_URI || "mongodb://localhost:27017/mydatabase",
  REDIS_HOST: Bun.env.REDIS_HOST || "127.0.0.1",
  REDIS_PORT: Number(Bun.env.REDIS_PORT) || 6379,
  REDIS_USERNAME: Bun.env.REDIS_USERNAME || "default",
  REDIS_PASSWORD: Bun.env.REDIS_PASSWORD || "my-top-secret",
  REDIS_DB: Number(Bun.env.REDIS_DB) || 0,
  ENABLE_DOCS:
    Bun.env.ENABLE_DOCS.toLowerCase() === "true" ||
    Bun.env.ENABLE_DOCS === "1" ||
    false,
};
